### PR TITLE
RISC0 Rust backend

### PR DESCRIPTION
**Pull request by lukaszcz**
* Adds a RISC0 backend which generates Rust code that can be compiled with the official RISC0 toolchain.
* The RISC0 backend is a wrapper around the Rust backend.
* Adds the `risc0-rust` to the `compile` CLI command, which creates a directory containing host and guest Rust sources for the RISC0 zkVM. The generated code can be compiled/run using `cargo` from inside the created directory (requires having RISC0 installed: https://dev.risczero.com/api/zkvm/install).

### CODE REVIEW

File: app/Commands/Compile/RiscZeroRust.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    rustRuntimeToml :: ByteString
+    rustRuntimeToml = $(FE.makeRelativeToProject "runtime/rust/juvix/Cargo.toml" >>= FE.embedFile)
+
+    writeDirFiles :: [(FilePath, ByteString)] -> Path Abs Dir -> Sem r ()
+    writeDirFiles fs outDir = do
+      let fs' = map (first relFile) fs
+      forM_ (first (outDir <//>) <$> fs') (uncurry writeFile)
+
+    writeFile :: Path Abs File -> ByteString -> Sem r ()
```
Comment (janmasrovira) [inline app/Commands/Compile/RiscZeroRust.hs:43]: This function seems generic enough to be defined somewhere else. Maybe in `Juvix.Data.Effect.Files`

File: app/Commands/Extra/NewCompile.hs [hunk 2]
```diff
@@ -27,6 +27,20 @@ getOutputFile ext inp = \case
...
+  Maybe (AppPath File) ->
+  Maybe (AppPath Dir) ->
+  Sem r (Path Abs Dir)
+getOutputDir ext inp = \case
+  Just out -> fromAppPathDir out
+  Nothing -> do
+    mainFile <- getMainFile inp
+    invokeDir <- askInvokeDir
+    let baseOutputDir = invokeDir <//> filename (replaceExtension' (fileExtToString ext) mainFile)
+    return $ absDir $ toFilePath baseOutputDir
```
Comment (janmasrovira) [inline app/Commands/Extra/NewCompile.hs:42]: having some helper function like `pathFileToPathDir : Path b File -> path b Dir` would make this more readable. The helper function can go to `Juvix.Prelude.Path`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -138,6 +139,13 @@ jobs:
         if: steps.cache-cairo-vm.outputs.cache-hit != 'true'
         uses: actions-rust-lang/setup-rust-toolchain@v1
 
+      - name: Install RISC0 VM
+        shell: bash
+        run: |
+          cargo install cargo-binstall
```
Comment (paulcadman) [inline .github/workflows/ci.yml:145]: Can this be cached? This can be done in a subsequent PR if possible.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -346,6 +354,13 @@ jobs:
...
```
Comment (paulcadman) [inline .github/workflows/ci.yml:360]: Can this be cached? This can be done in a subsequent PR if possible.

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**